packages/tabs/src/Tabs.ts

```diff
@@ -245,16 +245,55 @@ export class Tabs extends SizedMixin(Focusable, { noDefaultSize: true }) {
         return complete;
     }
 
-    public async scrollToSelection(): Promise<void> {
+    private getNecessaryAutoScroll(index: number): number {
+        const selectedTab = this.tabs[index];
+        const selectionEnd = selectedTab.offsetLeft + selectedTab.offsetWidth;
+        const viewportEnd = this.tabList.scrollLeft + this.tabList.offsetWidth;
+        const selectionStart = selectedTab.offsetLeft;
+        const viewportStart = this.tabList.scrollLeft;
+
+        // If the tab is already visible, the necessary scroll factor is 0.
+        let scrollTarget = 0;
+
+        if (selectionEnd > viewportEnd) {
+            // Selection is on the right side, not visible.
+            const nextTab = this.tabs[index + (this.dir === 'rtl' ? -1 : 1)];
+            scrollTarget = nextTab
+                ? nextTab.offsetLeft - this.tabList.offsetWidth
+                : viewportEnd;
+        } else if (selectionStart < viewportStart) {
+            // Selection is on the left side, not visible.
+            const prevTab = this.tabs[index + (this.dir === 'rtl' ? 1 : -1)];
+            const leftmostElement =
+                this.dir === 'rtl' ? -this.tabList.offsetWidth : 0;
+            scrollTarget = prevTab
+                ? prevTab.offsetLeft + prevTab.offsetWidth
+                : leftmostElement;
+        }
+
+        return scrollTarget;
+    }
+
+    public async maybeScrollToSelection(): Promise<void> {
         if (!this.enableTabsScroll || !this.selected) {
             return;
         }
 
         await this.updateComplete;
```

Comment:
Why this is required here?

Reply:
here I am waiting for `this.tabs` to be populated, i observed that unit tests failed without it because of this array being empty :( open to any other suggestion

Reply:
If we execute the next set of instructions only if `this.tabs` is truthy, will that work for you! This await will be flaky!

Reply:
Unfortunately no, the feature will not work at all without this await